src/content/React-CSharp/ClientApp/src/components/api-authorization/AuthorizeRoute.js

```diff
@@ -1,56 +1,46 @@
-import React from 'react'
-import { Component } from 'react'
-import { Route, Redirect } from 'react-router-dom'
-import { ApplicationPaths, QueryParameterNames } from './ApiAuthorizationConstants'
-import authService from './AuthorizeService'
+import React from 'react';
+import { useState, useEffect } from 'react';
+import { Route, Redirect } from 'react-router-dom';
+import { ApplicationPaths, QueryParameterNames } from './ApiAuthorizationConstants';
+import authService from './AuthorizeService';
 
-export default class AuthorizeRoute extends Component {
-    constructor(props) {
-        super(props);
+export function AuthorizeRoute(props) {
+  const [ready, setReady] = useState(false);
+  const [authenticated, setAuthenticated] = useState(false);
 
-        this.state = {
-            ready: false,
-            authenticated: false
-        };
-    }
+  useEffect(() => {
+    const populateAuthenticationState = async () => {
+      const authenticated = await authService.isAuthenticated();
+      setReady(true);
+      setAuthenticated(authenticated);
+    };
 
-    componentDidMount() {
-        this._subscription = authService.subscribe(() => this.authenticationChanged());
-        this.populateAuthenticationState();
-    }
+    const authenticationChanged = async () => {
+      setReady(false);
+      setAuthenticated(false);
+      await populateAuthenticationState();
+    };
 
-    componentWillUnmount() {
-        authService.unsubscribe(this._subscription);
-    }
+    const subscription = authService.subscribe(() => authenticationChanged());
+    populateAuthenticationState();
+    return () => authService.unsubscribe(subscription);
+  }, [])
 
-    render() {
-        const { ready, authenticated } = this.state;
-        var link = document.createElement("a");
-        link.href = this.props.path;
-        const returnUrl = `${link.protocol}//${link.host}${link.pathname}${link.search}${link.hash}`;
-        const redirectUrl = `${ApplicationPaths.Login}?${QueryParameterNames.ReturnUrl}=${encodeURIComponent(returnUrl)}`
-        if (!ready) {
-            return <div></div>;
-        } else {
-            const { component: Component, ...rest } = this.props;
-            return <Route {...rest}
-                render={(props) => {
+  const link = document.createElement("a");
+  link.href = props.path;
+  const returnUrl = `${link.protocol}//${link.host}${link.pathname}${link.search}${link.hash}`;
```

Comment:
nit: I feel like this should be done through [URL](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL)

Reply:
@nulladdict I think that next code can be simplier but I'm not sure that it's fully suitable equivalent for previous version:
```suggestion
  const returnUrl = new URL(props.path, window.location.origin);
```

Reply:
actually i'm not sure why this createElement trick came in the first place, before https://github.com/dotnet/spa-templates/commit/33439ac89839cf97e117a228937b1287f17f9c74 it used to be just `window.location.href` which should've been fine
i guess there might be something i am missing, so let's just not touch this

Reply:
Hi, @javiercn 
Could you please help me a little? 

I'm trying to figure out the code for generating a redirect link after authentication to help make it more clear. As I can see in https://github.com/dotnet/spa-templates/commit/33439ac89839cf97e117a228937b1287f17f9c74 you fixed problem with unused `props.path` in `AuthorizeRoute.js`. And I think that it's enough to replace it with `new URL` call. But if you know additional possible scenarios, could you elaborate on them?